ironfish/src/wallet/wallet.ts

```diff
@@ -484,9 +483,47 @@ export class Wallet {
       }
     })
 
-    for (const account of accounts) {
-      const shouldDecrypt = await this.shouldDecryptForAccount(blockHeader, account)
+    const shouldDecryptAccounts = (
+      await Promise.all(
+        accounts.map((a) =>
+          this.shouldDecryptForAccount(blockHeader, a).then(
+            (shouldDecrypt): [Account, boolean] => [a, shouldDecrypt],
+          ),
+        ),
+      )
+    )
+      .filter((v) => v[1])
+      .map((v) => v[0])
```

Comment:
I think this can be simplified with `AsyncUtils.filter`

Reply:
Yea I was thinking the same thing